apps/nowcasting-app/components/charts/use-format-chart-data.tsx

```diff
@@ -0,0 +1,117 @@
+import { useMemo } from "react";
+import { formatISODateString } from "../utils";
+import { ChartData } from "./remix-line";
+
+//sperate paste forcaste from furute forcast (ie: after selectedTime)
+const getForecastChartData = (
+  selectedTime: string,
+  fr?: {
+    targetTime: string;
+    expectedPowerGenerationMegawatts: number;
+  }
+) => {
+  if (!fr) return {};
+
+  if (
+    new Date(fr.targetTime).getTime() >
+    new Date(selectedTime + ":00.000Z").getTime()
+  )
+    return {
+      FORECAST: Math.round(fr.expectedPowerGenerationMegawatts),
+    };
+  else if (
+    new Date(fr.targetTime).getTime() ===
+    new Date(selectedTime + ":00.000Z").getTime()
+  )
+    return {
+      FORECAST: Math.round(fr.expectedPowerGenerationMegawatts),
+      PAST_FORECAST: Math.round(fr.expectedPowerGenerationMegawatts),
+    };
+  else
+    return {
+      PAST_FORECAST: Math.round(fr.expectedPowerGenerationMegawatts),
+    };
+};
+const useFormatChartData = ({
+  nationalForecastData,
+  pvRealDataAfter,
+  pvRealDataIn,
+  selectedTime,
+}: {
+  nationalForecastData?: {
+    targetTime: string;
+    expectedPowerGenerationMegawatts: number;
+  }[];
+  pvRealDataAfter?: {
+    datetimeUtc: string;
+    solarGenerationKw: number;
+  }[];
+  pvRealDataIn?: {
+    datetimeUtc: string;
+    solarGenerationKw: number;
+  }[];
+  selectedTime?: string;
+}) => {
+  const data = useMemo(() => {
+    if (
+      nationalForecastData &&
+      pvRealDataAfter &&
+      pvRealDataIn &&
+      selectedTime
+    ) {
+      const chartMap: Record<string, ChartData> = {};
+
+      const addDataToMap = (
+        dataPoint: any,
+        getDatetimeUtc: (dp: any) => string,
+        getPvdata: (dp: any) => Partial<ChartData>
+      ) => {
+        const pvData = getPvdata(dataPoint);
+        const datetimeUtc = getDatetimeUtc(dataPoint);
+        if (chartMap[datetimeUtc]) {
+          chartMap[datetimeUtc] = {
+            ...chartMap[datetimeUtc],
+            ...pvData,
+          };
+        } else {
+          chartMap[datetimeUtc] = {
+            datetimeUtc: formatISODateString(datetimeUtc),
+            ...pvData,
+          };
+        }
+      };
+
+      pvRealDataAfter.forEach((pva) =>
+        addDataToMap(
+          pva,
+          (db) => db.datetimeUtc,
+          (db) => ({
+            GENERATION_UPDATED: Math.round(db.solarGenerationKw / 1000),
```

Comment:
why you are using KW instead of data we get from api with MW direct ?

Reply:
does the api give it in MW ? i only see it in `solarGenerationKw `, where did you see it.

Reply:
you are right